editor/src/clj/editor/html_view.clj

```diff
@@ -239,41 +235,47 @@
-      (.load web-engine url))))
-
-(defn- update-web-view!
+(g/defnk load-resource! [^WebView web-view resource-node _node-id html]
   "Loads the url corresponding to the current resource.
 
   This will open the resource at the new location if we've moved it.
 
   If we somehow manage to browse away from the project to an external
   page, this will somewhat annoyingly bring us back."
-  [view-id project]
-  (let [web-view ^WebView (g/node-value view-id :web-view)
-        web-engine (.getEngine web-view)]
-    (when-let [resource-node (g/node-value view-id :resource-node)]
-      (load-resource! web-engine project (g/node-value resource-node :resource)))))
+  (when resource-node
+    (let [web-engine (.getEngine web-view)
+          resource   (g/node-value resource-node :resource)
+          project    (project/get-project resource-node)
+          url        (resource-url (get-http-server! project) resource)]
+      (.load web-engine url))))
 
 (defn- handle-location-change! [project view-id new-location]
   (if-let [new-resource-node
-           (if (string/starts-with? new-location (http-server/url (get-http-server! project)))
-             (let [resource-path (subs new-location (count (http-server/url (get-http-server! project))))]
-               (project/get-resource-node project resource-path))
-             nil)]
-    (g/transact (view/connect-resource-node view-id new-resource-node))
+           (when (string/starts-with? new-location (http-server/url (get-http-server! project)))
+             (let [resource-path (subs new-location (count (http-server/url (get-http-server! project))))
+                   resource-node (project/get-resource-node project resource-path)
+                   resource (g/node-value resource-node :resource)]
+               (when (resource/openable-in-view-type? resource :html)
+                 resource-node)))]
+    (g/transact [(g/connect new-resource-node :html view-id :html)
+                 (view/connect-resource-node view-id new-resource-node)])
     (log/warn :message (format "Moving to non-local url or missing resource: %s" new-location))))
 
+(g/defnode WebViewNode
+  (inherits view/WorkbenchView)
+  (property web-view WebView)
+  (input html g/Str)
+  (output load-resource g/Any :cached load-resource!)) 
+
 (defn make-view
   [graph ^Parent parent html-node opts]
   (let [project       (or (:project opts) (project/get-project html-node))
         web-view      (make-web-view project)
         web-engine    (.getEngine web-view)
         view-id       (g/make-node! graph WebViewNode :web-view web-view)
-        repainter     (ui/->timer 1 "update-web-view!" (fn [_ _ _] (update-web-view! view-id project)))]
+        repainter     (ui/->timer 30 "update-web-view!" (fn [_ _ _] 
+                                                          (when (.isSelected (:tab opts))
+                                                            (g/node-value view-id :load-resource))))]
```

Comment:
Increasing the frames here makes the update more responsive. Since we also check if the tab is selected, that shouldn.t be a problem. Maybe there is a better way to handle this?